xbmc/cores/AudioEngine/AESinkFactory.h

```diff
@@ -38,12 +38,14 @@ struct AESinkInfo
 
 typedef IAESink* (*CreateSink)(std::string &device, AEAudioFormat &desiredFormat);
 typedef void (*Enumerate)(AEDeviceInfoList &list, bool force);
+typedef void (*Dispose)();
```

Comment:
maybe your first suggestion "cleanup" fits better here because we don't call it when the disk is disposed but when ae is shut down.